docs/developer_docs.md

```diff
@@ -0,0 +1,164 @@
+# *opencv-rueppelli-2*
+
+## developer documentation
+
+[detect circle](#detectCircle)
+
+[remove background](#remove background) 
+
+[sortings](#sortings)
+
+### detect circle
+
+int detectCircle(cv::Mat &image)
+   - input is an image, 
+   - returntype a number of found circles on the input image
+
+### remove background
+
+- cv:: Mat removeLightWithDifference(cv:: Mat image, cv:: Mat pattern)
+   - What does this program do?
+     - to extract the image from pattern
+   - Parameters
+     - image: Input image
+     - pattern: Input image with pattern
+   - returntype a new image
```

Comment:
this returntye thing is need to be fixed.
First it is return type.
Second it's return type is `cv::Mat`.
Retruns a new image with some features added...

Reply:
@rapcsakandi I am still not sure about that `returntype` thing. Return type would be something like `int` or `cv::Mat`